geoindex: skip certain tests for ARM64

otan opened the pull request:
These tests fail on the ARM64 build as they produce a slightly different result. As such, skip them for now (we have evidence of GEOS doing this for other things like VoronoiPolygons).

Refs: #106954
Resolves #106964

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106966)
<!-- Reviewable:end -->

otan committed:
geoindex: skip certain tests for ARM64

These tests fail on the ARM64 build as they produce a slightly different
result. As such, skip them for now (we have evidence of GEOS doing this
for other things like VoronoiPolygons).

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106966#-:-N_dDWpM7pmvjg8Yndyo:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
